### ISSUE TITLE
IDraggable: Add multi-axis support

### ISSUE THREAD

**Issue by BambOoxX** [ScottPlot/ScottPlot#1545]
## Features Suggestion

**Feature description:**

The mouse interaction for dragable plottables should take the axis index into account. Currently it seems to call
```cs
        /// <summary>
        /// Return the X position (in coordinate space) for the given pixel column
        /// </summary>
        /// <param name="xPixel">horizontal pixel location</param>
        /// <returns>horizontal position in coordinate space</returns>
        public double GetCoordinateX(float xPixel) => settings.XAxis.Dims.GetUnit(xPixel);

        /// <summary>
        /// Return the Y position (in coordinate space) for the given pixel row
        /// </summary>
        /// <param name="yPixel">vertical pixel location</param>
        /// <returns>vertical position in coordinate space</returns>
        public double GetCoordinateY(float yPixe) => settings.XAxis.Dims.GetUnit(yPixel);
```
Which only works if everything happens on the first axis (X and/or Y). If the plottable is on another set of axes, the `IsUnderMouse` and `DragTo` methods fail because the coordinates are wrong.

A short term patch could be to consider the second axes too with e.g. 
```cs
        /// <summary>
        /// Return the X position (in coordinate space) for the given pixel column
        /// </summary>
        /// <param name="xPixel">horizontal pixel location</param>
        /// <returns>horizontal position in coordinate space</returns>
        public double GetCoordinateX(float xPixel,int axisindex = 0) => axisindex == 0 ? settings.XAxis.Dims.GetUnit(xPixel) : settings.XAxis2.Dims.GetUnit(xPixel);

        /// <summary>
        /// Return the Y position (in coordinate space) for the given pixel row
        /// </summary>
        /// <param name="yPixel">vertical pixel location</param>
        /// <returns>vertical position in coordinate space</returns>
        public double GetCoordinateY(float yPixel,int axisindex = 0) => axisindex == 0 ? settings.XAxis.Dims.GetUnit(yPixel) : settings.YAxis2.Dims.GetUnit(yPixel);
```

If one adds other axes I am not sure how to retrieve the current one...

**Comment by BambOoxX** [comment]
In addition, to capture the current `IDraggable` object with GetDraggableUnderMouse, `Plot.GetCoordinateX` and `Plot.GetCoordinateY` are directly called too, resulting in the mouse failing to capture the `IDraggable` object.
```
        private IDraggable GetDraggableUnderMouse(double pixelX, double pixelY, int snapDistancePixels = 5)
        {
            double snapWidth = Settings.XAxis.Dims.UnitsPerPx * snapDistancePixels;
            double snapHeight = Settings.YAxis.Dims.UnitsPerPx * snapDistancePixels;

            foreach (IDraggable draggable in GetDraggables())
                if (draggable.IsUnderMouse(Plot.GetCoordinateX((float)pixelX), Plot.GetCoordinateY((float)pixelY), snapWidth, snapHeight))
                    if (draggable.DragEnabled)
                        return draggable;

            return null;
        }
```

**Comment by BambOoxX** [comment]
@swharden Do you think it is relevant to add the small patch or do you prefer an entire upgrade of this ?

**Comment by swharden** [comment]
@BambOoxX I've been attacking open issues latest-to-newest so it might take me a little bit to get to this one, but if you fix it and make a pull request I'd be happy to prioritize that and merge it in quickly!

**Comment by BambOoxX** [comment]
@swharden I am a bit rusty with the PRs but I will try to propose something.

**Comment by BambOoxX** [comment]
@swharden I just tried the PR, tell me if you see something wrong. Cheers !

**Comment by BambOoxX** [comment]
@swharden  I did not pay attention, but I believe #1556 should also fix #1536
